django/river/services.py

```diff
@@ -8,36 +8,29 @@
 from pydantic import ValidationError
 from river import models
 from river.adapters.event_publisher import EventPublisher
-from river.adapters.mappings import MappingsRepository
-from river.adapters.pyrog_client import PyrogClient
 from river.adapters.topics import TopicsManager
 from river.common.analyzer import Analyzer
 from river.common.database_connection.db_connection import DBConnection
 from river.domain.events import BatchEvent
 from river.extractor.extractor import Extractor
+from river.parsing import Source, as_old_mapping
 from river.transformer.transformer import Transformer
 from utils.json import CustomJSONEncoder
 
 
 def batch(
-    batch_instance: models.Batch,
-    resources: List[str],
+    batch_id: str,
+    resources: List[dict],
```

Comment:
Why do we need `resources`? Can't we retrieve them from `batch_id`?

Reply:
we retrieve them directly from the mappings we send to the batch route